core/commands/corenet.go

```diff
@@ -0,0 +1,176 @@
+package commands
+
+import (
+	"io"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+	corenet "github.com/ipfs/go-ipfs/core/corenet"
+
+	manet "gx/ipfs/QmPpRcbNUXauP3zWZ1NJMLWpe4QnmEHrd2ba2D3yqWznw7/go-multiaddr-net"
+	pstore "gx/ipfs/QmQdnfvZQuhdT93LNc5bos52wAmdr3G2p6G8teLJMEN32P/go-libp2p-peerstore"
+	ma "gx/ipfs/QmYzDkkgAEmrcNzFCiYo6L1dTX4EAG1gZkbtdbd9trL4vd/go-multiaddr"
+	context "gx/ipfs/QmZy2y8t9zQH2a1b8q2ZSLKp17ATuJoCNxxyMFG5qFExpt/go-net/context"
+)
+
+var CorenetCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Identity based p2p data transfer",
+	},
+
+	Subcommands: map[string]*cmds.Command{
+		"listen": listenCmd,
+		"dial":   dialCmd,
+	},
+}
+
+var listenCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Start listening for incoming corenet connections",
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("Handler", true, false, "Address of application handling the connections"),
+		cmds.StringArg("Protocol", true, false, "Protocol name"),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		if !n.OnlineMode() {
+			res.SetError(errNotOnline, cmds.ErrClient)
+			return
+		}
+
+		malocal, err := ma.NewMultiaddr(req.Arguments()[0])
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		listener, err := corenet.Listen(n, "/app/"+req.Arguments()[1])
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		defer listener.Close()
+
+		for {
+			remote, err := listener.Accept()
```

Comment:
I haven't found a way to get peer address. Should we left determining peer identity up to applications using this interface or is there actually a way to get peer ID?